File: april/10.py
n,m = map(int,input().split())
A = list(map(int,input().split()))
B = list(map(int,input().split()))
B.sort()
p = set([(B[0] - a + m)%m for a in A])

for x in p:
    tmp = [(a+x)%m for a in A]
    tmp.sort()
    if tmp == B:
        print(x)
        break

File: april/10.cpp
#include<bits/stdc++.h>
using namespace std;

int main()
{
    int n,m;
    cin>>n>>m;
    vector<int> A(n),B(n);
    for(int i =0;i<n;i++) cin>>A[i];
    for(int i =0;i<n;i++) cin>>B[i];
    set<int> candidates;
    for(int i =0;i<n;i++) candidates.insert((B[0] - A[i])%m);

    sort(B.begin(),B.end());
    int f;
    for(int x:candidates)
    {
        f =1;
        vector<int> tmp = A;
        for(int i=0;i<n;i++)
        {
            tmp[i] = (tmp[i]+x)%m;
        }

        sort(tmp.begin(),tmp.end());

        if(tmp == B)
        {
            cout<<x<<endl;
            f=0;
            break;
        }

    }

    if (f)
        cout<<4<<endl;
}